### ISSUE TITLE
[Bug] Multiple problems on UWP after upgrading SR5 to SR6 (blank page instead of content, buttons without text, and others)

### ISSUE THREAD

**Issue by holecekp** [xamarin/Xamarin.Forms#14811]
### Description
I have upgraded Xamarin.Forms 5.0 SR5 to SR6. Everything is all right on Android. However, UWP version seems to be completely broken. I am sorry that I am posting just a brief summary instead of creating a separate issue for each of the bugs but it would be very time consuming to describe all of them in detail, to find the exact conditions when they happen, and to create a repro project for each of them.

**Problem 1: Blank page is displayed after navigation**
I am navigating to one page using the following code:
```csharp
var myPage = new MyPage(viewModelForMyPage);
await Shell.Current.Navigation.PushModalAsync(myPage);
```
SR5 navigates to the new page correctly. SR6 displays the new page correctly only on Android and it displays a blank page on UWP. The content of the new page does not matter because I have replaced it with a single label and still no content is displayed. The user is navigated to the new page (I have added a break point to the constructor and another one to OnAppearing() and both are triggered) just the content is not rendered.

It is interesting that another page uses the same mechanism to navigate to yet another page and it works even in SR6.

**Problem 2: Buttons are sometimes invisible**
A button is sometimes not displayed. This is very strange and random issue. The problematic button has IsVisibility with a binding to the view model. It is invisible initially. When it should be shown, sometimes it is (about 70 %) of cases, and sometimes it is not. Another strange issue is that when some button is not displayed because of this bug, redrawing page after resizing the window becomes also broken. When the user resizes the window, nothing happens and the content in the window is not adjusted.

**Problem 3: Other content is sometimes also not displayed**
Similar problem is also with other components (ImageView, Label). The common property is that the components are not initially visible and they are shown and hidden using a binding in the IsVisible property.

**Problem 4: Controls are sometimes rendered on a wrong position**
This is again a random issue. Sometimes the page is displayed correctly. Sometimes the same page is displayed but with the controls on wrong positions.
![ButtonWrongPosition](https://user-images.githubusercontent.com/24872113/139596297-62d08646-5f77-4d0d-b95f-9c56b20516e2.png)

**Problem 5: Buttons are sometimes displayed but without any text**
This is a less common random issue. Sometimes the button is displayed but there is not any text in it. The texts are in RESX files and they are not modified during the run-time. Also in this case, page redrawing after resizing the window stops working. The buttons looks like this:
```xml
<Button
   Text="{Static resources:Strings.NextQuestion}"
    IsVisible=Binding ShowNextQuestionButton}" />
```
![ButtonNoText](https://user-images.githubusercontent.com/24872113/139596472-80b8fee5-1cf5-4e02-b3e7-3ae748e277ca.png)

### A few other remarks
- I have tried to clean the solution and rebuild it. I have also tried to delete all bin and obj folders.
- I am using Shell.
- I am using lazy views to improve slow loading times. It is my slightly modified version of LazyView from Xamarin Community Toolkit which is initialized automatically when IsVisible is changed from false to true for the first time. It should not matter for the Problem 1 (blank page). The not-displayed buttons are also on pages that do not use LazyViews.
- The changes log for SR6 mentions fixes for buttons in CollectionView on UWP but these problematic buttons are not in CollectionView.
- Downgrading back to SR5 solves all the problems immediately.

### Basic
 Information

- Version with issue: 5.0.0.2196
- Last known good version: 5.0.0.2125
- Platform Target Frameworks: 
  - UWP:  19041
- NuGet Packages: Xamarin.Forms, Essentials, Xamarin Community Toolkit, AppCenter, System.Text.Json

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
Microsoft Visual Studio Community 2019
Version 16.11.5
VisualStudio.16.Release/16.11.5+31729.503
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

Visual C++ 2019   00435-60000-00000-AA088
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.11.75.64347
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.11.75.64347
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.11.75.64347
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.11.75.64347
Azure Functions and Web Jobs Tools

C# Tools   3.11.0-4.21403.6+ae1fff344d46976624e68ae17164e0607ab68b10
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.10.15 (552afdf)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.11.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Razor (ASP.NET Core)   16.1.0.2122504+13c05c96ea6bdbe550bd88b0bf6cdddf8cde1725
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   16.0.62107.28140
Microsoft SQL Server Data Tools

TypeScript Tools   16.0.30526.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.11.0-4.21403.6+ae1fff344d46976624e68ae17164e0607ab68b10
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.11.0-beta.21322.6+488cc578cafcd261d90d748d8aaa7b8b091232dc
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.11.000.190 (d16-11@2391ed9)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.11.0.17 (remotes/origin/11e0001f0b17269345e80b58fb3adf1ba4efe2cd@11e0001f0)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.10.5 (355b57a)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.4.0.5 (d16-11/7776c9f)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: c633fe9
    Java.Interop: xamarin/java.interop/d16-11@48766c0
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.35.4@85460d3
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@683f375


Xamarin.iOS and Xamarin.Mac SDK   15.0.0.8 (0796d78dc)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

</details>

### Workaround

Downgrade to SR5

**Comment by MitchBomcanhao** [comment]
seems like this issue, which will have been fixed whenever SR7 is out - [[Bug] [Regression] UWP Button breaks the Layout](https://github.com/xamarin/Xamarin.Forms/issues/14764)

**Comment by jfversluis** [comment]
Thanks @MitchBomcanhao ! Indeed there were some things broken with SR6 for UWP, sorry about that. @holecekp would you be maybe able to try out the nightly version (see the README) and see if that fixes things or report what still seems broken?

Thanks!

**Comment by holecekp** [comment]
@jfversluis I have tested the nightly version briefly and the good news that I have found none of these UWP issues in it. So they will be fixed when SR7 is out.

The bad news is that the nightly version is currently unusable on Android. It introduced a serious bug that there are just black boxes instead of some controls. Because it is not a finished stable version, I assume that you know about problem and it will be fixed before the release so I will not open a new issue for it.
